### PR TITLE
Color Palette SPI changes

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA: TNT-933

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9195/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11022/ : FAILURE in 4m 09s

**Comment by oleg-morozov** [comment]
extended test

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by oleg-morozov** [comment]
extended check plugins

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9201/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11028/ : FAILURE in 12m 32s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-dashboard-plugin-tests/30/ : SUCCESS in 12m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9205/ : SUCCESS in 7s

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**headlineChartDrilling.spec.ts**|2|-|9.1220s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|12.7400s|
|✅|**attributeFilterConfig.spec.ts**|3|-|31.7700s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|13.0710s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|23.3130s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|19.5830s|
|✅|**barChartDrilling.spec.ts**|4|-|12.4270s|
|✅|**pivotTableSizing.spec.ts**|1|-|10.9690s|
|✅|**dateFilter.spec.ts**|9|-|20.8870s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|13.1620s|
|✅|**dashboardFiltering.spec.ts**|2|-|27.1020s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|44.3640s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|12.0090s|
|✅|**dashboard.spec.ts**|9|-|55.0020s|
|✅|**attributeFilterButton.spec.ts**|9|-|37.8240s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|25.8370s|
|✅|**drilling.spec.ts**|2|-|23.6900s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/273/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9204/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/273/ : SUCCESS in 15m 52s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2836

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9203/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1740/ : SUCCESS in 17m 52s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1787/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9208/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11031/ : SUCCESS in 19m 24s

File: libs/sdk-backend-base/src/dummyBackend/index.ts [hunk 1]
```diff
@@ -454,40 +456,52 @@ class DummyOrganization implements IOrganization {
...
-                    ref: idRef("theme_id"),
-                    uri: "theme_uri",
-                    theme: theme.theme,
-                });
-            },
+            createTheme: resolveTheme,
+            updateTheme: resolveTheme,
             deleteTheme: () => Promise.resolve(),
+            getColorPalettes: () => Promise.resolve([]),
+            getActiveColorPalette: () => Promise.resolve(undefined),
```
Comment (ivan-nejezchleb) [inline libs/sdk-backend-base/src/dummyBackend/index.ts:499]: undefined not needed

**Review by scavnickyj — APPROVED** [review]

File: libs/sdk-backend-base/src/dummyBackend/index.ts (hunk 1 above)
Comment (scavnickyj) [reply libs/sdk-backend-base/src/dummyBackend/index.ts:499]: maybe remove it from the theme as well

Comment (oleg-morozov) [reply libs/sdk-backend-base/src/dummyBackend/index.ts:499]: TypeScript gives an error without it.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2836

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2611/ : SUCCESS in 1m 35s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2910/ : SUCCESS in 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1743/ : SUCCESS in 18m 52s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11038/ : SUCCESS in 22m 04s

**Pull request merged**